pubpub/platform
# dev: upgrade vitest and properly set environment per type of test

- **dev: upgrade vitest related dependencies and setup proper environments for db tests-n**

## Issue(s) Resolved
Replaces weird and unmaintained `vitest-tsconfig-paths` with `vite-tsconfig-paths`.

## High-level Explanation of PR

While trying to fix a type error from `vitest-tsconfig-paths` in #646, I realized this packages is literally just an old version of `vite-tsconfig-paths`.

I installed `vite-tsconfig-paths` instead, which has proper types. This also encouraged an upgrade to vite 2.

While doing this upgrade, I realized that tests now would fail bc our environment variable validation treats the database tests as client tests, as we are running in the `jsdom` environment.

Rather than just turn off validation, I thought it better to just set the environment properly.

Database tests now run in the `node` environment if their file is like `*.db.test.ts`.


## Test Plan
1. Run `pnpm --filter core vitest`
2. See no errors.

## Screenshots (if applicable)

## Notes
